sys/decl.go

```diff
@@ -236,10 +236,21 @@ func (t FilenameType) Align() uintptr {
 	return 1
 }
 
+type ArrayKind int
+
+const (
+	ArrayRandLen ArrayKind = iota
+	ArrayFixedLen
```

Comment:
I think it's better to have just 2 kinds: rand and range. Fixed len can be represented by a degenerate range of [X:X], and then it should be handled automatically by existing code that works with ranges. This way we will have fewer cases in switches, lower possibility of bugs and better tested code.